src/core/net/nat64_translator.cpp

```diff
@@ -99,7 +99,10 @@ Error Translator::SendMessage(Message &aMessage)
 
     VerifyOrExit(result == kForward);
 
-    error = Get<Ip6::Ip6>().SendRaw(aMessage, !OPENTHREAD_CONFIG_IP6_ALLOW_LOOP_BACK_HOST_DATAGRAMS);
+    // Do the opposite of OPENTHREAD_CONFIG_IP6_ALLOW_LOOP_BACK_HOST_DATAGRAMS.
+    aMessage.SetOrigin(OPENTHREAD_CONFIG_IP6_ALLOW_LOOP_BACK_HOST_DATAGRAMS ? Message::Origin::kFromHostDisallowLoopBack
```

Comment:
I think we should actually preserve the value (OT_MESSAGE_ORIGIN_FROM_HOST_UNTRUSTED) from https://github.com/openthread/openthread/pull/9437/files#diff-78c7ba9f144904d8f4b0fe8798af649fdf51e8c81e98cfd26e6791b488b34801R1146 ?

An IPv6 packet translated from IPv4 should also be untrusted since we have almost no assumptions on the exact behavior of the NAT44 on the platform (except that it is a stateful NAT44).

Reply:
Good point.